### ISSUE TITLE
Post-install tests for ML libraries

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#31552]
Spack offers post-install tests for packages that can be run like:
```console
$ spack test run py-torch
```
These tests are designed to serve as smoke tests to ensure that installations work as expected. All Python packages already have import tests, but we can do better than this. I propose that we add tests to all ML libraries in Spack that would (for example):

1. Instantiate a small neural network (single layer MLP)
2. Push the model to whatever backend we build for (CUDA, ROCm, MPS, etc.)
3. Pass data through the model and compute loss
4. Backpropagate the loss

For PyTorch, this would be a pruned down version of their [Quickstart](https://pytorch.org/tutorials/beginner/basics/quickstart_tutorial.html) tutorial that uses fake data instead of downloading MNIST.

Tracking which packages need tests:

- [ ] PyTorch
- [ ] TensorFlow
- [ ] scikit-learn
